src/app/modules/main/browser_section/bookmark/model.nim

```diff
@@ -0,0 +1,118 @@
+import NimQml, Tables, strutils, stew/shims/strformat
+
+import item
+
+type
+  ModelRole {.pure.} = enum
+    Name = UserRole + 1
+    Url = UserRole + 2
+    ImageUrl = UserRole + 3
+
+QtObject:
+  type
+    Model* = ref object of QAbstractListModel
+      items: seq[Item]
+
+  proc delete(self: Model) =
+    self.items = @[]
+    self.QAbstractListModel.delete
+
+  proc setup(self: Model) =
+    self.QAbstractListModel.setup
+
+  proc newModel*(): Model =
+    new(result, delete)
+    result.setup
+
+  proc `$`*(self: Model): string =
+    for i in 0 ..< self.items.len:
+      result &= fmt"""
+      [{i}]:({$self.items[i]})
+      """
+
+  proc modelChanged(self: Model) {.signal.}
```

Comment:
Pretty sure this signal is not needed. Looks like bad understanding of models by me from back in the day :sweat_smile:

Reply:
https://github.com/status-im/status-desktop/issues/18556